Browser_IDE/executionEnvironment_Page.js

```diff
@@ -80,34 +91,44 @@ document.getElementById("canvas").addEventListener("click", async function () {
 
 // Convenience function for reporting errors, printing them to the terminal
 // and also sending a message to the main window.
-function ReportError(block, message, line, formatted=false){
+function ReportError(block, message, line, stacktrace ,formatted=false){
     let outputMessage = message != "";
+    let stackTrace = stacktrace;
+
+    // Escape only the user-provided input
+    let escapedBlock = escapeHtml(block);
+    let escapedMessage = escapeHtml(message);
 
     if (outputMessage && line != null)
-        message = "Error on line "+line+": "+message;
+        escapedMessage = "Error on line "+line+": "+escapedMessage;
 
-    if (block != null && block != "" && block != "__USERCODE__null") {
-        if (!block.startsWith(userCodeBlockIdentifier)){
-            message = "Please file a bug report and send us the following info!\n    Error in file: "+block+"\n    "+message;
-            block = "Internal Error";
+    if (escapedBlock != null && escapedBlock != "" && escapedBlock != "__USERCODE__null") {
+        if (!escapedBlock.startsWith(userCodeBlockIdentifier)){
+            escapedMessage = "Please file a bug report and send us the following info!\n    Error in file: "+escapedBlock+"\n    "+escapedMessage;
+            escapedBlock = "Internal Error";
         }
         else{
-            block = block.slice(userCodeBlockIdentifier.length);
+            escapedBlock = escapedBlock.slice(userCodeBlockIdentifier.length);
         }
         if (outputMessage)
-            message = "(" + block + ") " + message;
+            escapedMessage = "(" + escapedBlock + ") " + escapedMessage;
     }
+    
+    // Format the stack trace with <details> and <summary> tags
+    escapedMessage = '<summary style="color: red;">' + escapedMessage + '</summary>';
+    stackTrace = '<pre>' + stackTrace + '</pre>';
 
     if (outputMessage && !formatted)
-        message = "\x1b[0m\x1b[31m" + message + "\x1b[0m";
+        escapedMessage = '<details>' + escapedMessage + stackTrace + '</details>';

```

Comment:
For this bit here, just a few small fixes to improve the results on the C++ language side
1. If `formatted` is true, make sure not to add the `color: red` styling
2. Make sure to check if the stackTrace is empty - if it is, make it output _without_ the `summary` wrapper

Also not sure where to write this, but make sure to also update the use of ReportError in runtimes/ExecutionEnvironmentInternal.js.